step/step.go

```diff
@@ -61,6 +62,12 @@ func (d IntermediateFileDownloader) ProcessConfig() (Config, error) {
 	stepconf.Print(input)
 	d.logger.EnableDebugLog(input.Verbose)
 
+	if strings.TrimSpace(string(input.BitriseAPIAccessToken)) == "" {
+		d.logger.Warnf("Bitrise API token is unavailable, skipping step.")
+		d.logger.Printf("Hint: maybe this build is running in the first stage of a pipeline? If so, pulling intermediate files is not possible.")
+		os.Exit(0)
+	}
```

Comment:
Everywhere else this `ProcessConfig` processes the input and either returns a valid config struct or an error. This a bit goes against that by exiting early which might be a surprise if you are not aware of it. 

What if we have a dedicated error for this and return that error. Then move this logging logic into the main function instead. This way the logic would not be that hidden / unexpected. So something like this:
```
config, err := downloader.ProcessConfig()
if err != nil {
    if err == downloader.ErrMissingAccessToken {
        d.logger.Warnf("Bitrise API token is unavailable, skipping step.")
 	d.logger.Printf("Hint: maybe this build is running in the first stage of a pipeline? If so, pulling intermediate files is not possible.")
        return 0       
    }
    
    logger.Errorf("Process config: %s", err)
    return 1
}
```

Reply:
Good idea, take a look at the latest commit